lib/runner.py

```diff
@@ -1,83 +1,31 @@
-import pandas as pd
+from lib.adapters.base import Credentials
+from lib.adapters import ADAPTERS
+from lib.exceptions import UnknownExchange
 
+from strategy import Strategy
 from account import Account
-from binance.client import Client
-from binance.websockets import BinanceSocketManager
 
 
 class Runner(object):
     """
-        The Runner is responsible for handling any exchange interactions,
-        and running a given `Strategy` against the data that is provided
-        from the exchange.
-
-        A `Strategy` is expected to have two methods (a) `start()` - which
-        performs any initialisation - such as configuring instance attributes,
-        and (b) `process(runner, tick_dataframe)`, which computes any actions
-        to run based upon the incoming data from the exchange.
+    The Runner is responsible for configuring all components required to execute
+    a trading algorithm. It's called by the entrypoint of the app, located in
+    `run.py`.
     """
 
-    def __init__(self, apiKey, apiSecret, symbol, runnable):
-        self.client = Client(apiKey, apiSecret)
-
-        account_info = self.client.get_account()
-        self.account = Account(account_info)
-
-        self.bm = BinanceSocketManager(self.client)
-        self.bm.start_user_socket(self.account)
+    def __init__(self, creds: Credentials, symbol:str, strategy: Strategy):
```

Comment:
This method is the *last point* at which we can simply fail and terminate without any clean-up, as there's no connections in progress yet. Something worth thinking about.